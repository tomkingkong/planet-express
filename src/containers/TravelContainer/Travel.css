.Travel {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: relative;
  height: calc(100vh - 65px);
  width: 100%;
  margin: 0 auto; }

.travel__locations {
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-end;
  align-self: flex-end;
  border-top: 0.1rem solid #7ed2d5;
  border-bottom: 0.2rem solid #7ed2d5;
  overflow: scroll;
  height: 100%;
  width: 100%;
  transition: all 0.4s ease-out; }

.travel__title {
  color: white;
  font-size: 140%;
  text-align: center;
  z-index: 9999;
  height: 3rem;
  width: 100%;
  padding: 0.5rem 0;
  transition: all 0.1s linear; }
  .travel__title h4 {
    top: 0;
    font-family: Trench;
    font-weight: 600;
    letter-spacing: 1px;
    margin: 0; }

.smaller {
  box-shadow: 0 0.1rem 1rem 0.01rem rgba(86, 88, 96, 0.872);
  font-size: 100%;
  height: 6%; }

.loading {
  height: 0;
  background: url("/images/loading/loading_mini-spaceship-resources.gif") no-repeat center;
  background-position: 100%; }
