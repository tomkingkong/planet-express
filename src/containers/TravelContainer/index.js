import React, { Component } from 'react';
import { array, object } from 'prop-types';
import { connect } from 'react-redux';

import { handleScroll } from '../../utilities/ScrollAnimate';
import './Travel.css';
import DestinationCard from '../DestinationCard';

export class TravelContainer extends Component {
  displayExoplanets = () => {
    const { exoplanets, history } = this.props;
    return exoplanets.map((planet, i) => (
      <DestinationCard 
        key={i} 
        planet={planet} 
        history={history} 
      />
    ));
  }

  render() {
    handleScroll('.travel__locations', '.travel__title');
    return (
      <div className="Travel">
        <header className="travel__title">
          <h4>Available Destinations</h4>
        </header>
        <section className="travel__locations">
          { this.displayExoplanets() }
        </section>
      </div>
    );
  }
}

TravelContainer.propTypes = {
  exoplanets: array,
  history: object
};

export const mapStateToProps = ({ exoplanets }) => ({ exoplanets });

export default connect(mapStateToProps, null)(TravelContainer);