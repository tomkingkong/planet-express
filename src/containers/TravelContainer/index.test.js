import React from 'react';
import { shallow } from 'enzyme';

import { TravelContainer, mapStateToProps } from '../TravelContainer';

describe('Travel Container', () => {
  const mockPlanets = [
    {
      name: 'HD 16141 b',
      mass: 0.26,
      image: '/images/alternative/exoplanet_rocky_5.jpg',
      distance: 35.91,
      numInSystem: '1st',
      yearLength: 75.523,
      hostStar: 'HD 16141',
      numMoons: 0,
      discoveryBy: 'W. M. Keck Observatory',
      discoveryMethod: 'Radial Velocity'
    }
  ]
  it('should match snapshot', () => {
    const wrapper = shallow(<TravelContainer exoplanets={[{}]} history={{}}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with planet data', () => {
    const wrapper = shallow(<TravelContainer exoplanets={mockPlanets} history={{}}/>);
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should receive an exoplanets array from props', () => {
      const mockState = {
        exoplanets: [ {planet: 1} ],
        technologies: [],
        spaceEvents: [],
        journey: {},
        news: []
      };
      const expected = {exoplanets: [ {planet: 1} ]};

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
});