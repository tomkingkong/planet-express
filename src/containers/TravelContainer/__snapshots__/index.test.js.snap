// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Travel Container should match snapshot with planet data 1`] = `
<div
  className="Travel"
>
  <header
    className="travel__title"
  >
    <h4>
      Available Destinations
    </h4>
  </header>
  <section
    className="travel__locations"
  >
    <Connect(DestinationCard)
      history={Object {}}
      key="0"
      planet={
        Object {
          "discoveryBy": "W. M. Keck Observatory",
          "discoveryMethod": "Radial Velocity",
          "distance": 35.91,
          "hostStar": "HD 16141",
          "image": "/images/alternative/exoplanet_rocky_5.jpg",
          "mass": 0.26,
          "name": "HD 16141 b",
          "numInSystem": "1st",
          "numMoons": 0,
          "yearLength": 75.523,
        }
      }
    />
  </section>
</div>
`;

exports[`Travel Container should match snapshot without data 1`] = `
<div
  className="Travel"
>
  <header
    className="travel__title"
  >
    <h4>
      Available Destinations
    </h4>
  </header>
  <section
    className="travel__locations loading"
  />
</div>
`;
